fs/dtype_linux_test.go

```diff
@@ -42,7 +43,7 @@ func testSupportsDType(t *testing.T, expected bool, mkfs ...string) {
 		t.Fatal(err)
 	}
 	t.Logf("Supports d_type: %v", result)
-	assert.Equal(t, expected, result)
```

Comment:
Can we have an equal and deep equal function in test yourself? Check + is. Equal is awkward.

Reply:
[`assert.Equal()`](https://godoc.org/github.com/gotestyourself/gotestyourself/assert#Equal) exists. It's a convenience function for `assert.Assert(t, is.Equal())`.

This was migrated using an [automated tool](https://github.com/gotestyourself/gotestyourself/pull/36) which preserves the existing behaviour. Since this was using assert (Errorf/Fail) it kept that behaviour by using `assert.Check()`. 

I can manually change some of these to use `assert.Equal()` when it's correct, but it's not always correct to do so (ex: when in a loop for a test table, or when checking multiple return values).

Separating the failure (Assert -> FailNow vs Check -> Fail) from the comparison logic is the only way to avoid the duplication. However I agree that adding convenience functions for the most common cases is a good idea. Two already exist (Equal and NilError).

Right now "DeepEqual" is called `Compare()` because I felt that `DeepEqual` implied `reflect.DeepEqual`, which is not how it's implemented. So I kept it with `Compare` to match the library name. Do you have an opinion on that terminology?

Reply:
I would prefer having an equivalent to `reflect.DeepEqual`. It has a standard definition and mostly does what one needs when `==` isn't enough.

The issue here is that the sugar here is cognitively inefficient. The idiomatic usage requires not just looking up the right function, but clicking through to the documentation to understand what its doing and to use it. For example, `Compare` classically has always returned a partial ordering.

Reply:
`reflect.DeepEqual` is not correct for this use case:
1. it always compares unexported fields. This means that an implementation change in an unrelated package can cause a test to fail. `go-cmp` requires that you are explicit about either ignoring or allowing unexported fields for each type in the comparison that has unexported fields
2. it does not provide any way to customize comparisons for values you can't control (like timestamps) this means you are forced to do a lot more work to either ignore values (by zero'ing them out), or you lose the ability to use a DeepEquals entirely.

`go-cmp` was created specifically to handle these cases. It behaves the same way as `DeepEqual` otherwise.

> Compare classically has always returned a partial ordering.

I don't know what you mean by this. Could you provide an example?

Reply:
Are you referring to a comparison function for sort, that would return (-1, 0, 1) ?